span.go

```diff
@@ -8,39 +8,52 @@ package datadog
 import (
 	"encoding/binary"
 	"fmt"
+	"net/http"
 	"strconv"
 
 	"go.opencensus.io/trace"
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
 )
 
-// canonicalCodes maps (*trace.SpanData).Status.Code to their description. See:
+// statusCodes maps (*trace.SpanData).Status.Code to their message and http code. See:
 // https://github.com/googleapis/googleapis/blob/master/google/rpc/code.proto.
-var canonicalCodes = [...]string{
-	"ok",
-	"cancelled",
-	"unknown",
-	"invalid_argument",
-	"deadline_exceeded",
-	"not_found",
-	"already_exists",
-	"permission_denied",
-	"resource_exhausted",
-	"failed_precondition",
-	"aborted",
-	"out_of_range",
-	"unimplemented",
-	"internal",
-	"unavailable",
-	"data_loss",
-	"unauthenticated",
+var statusCodes = map[int32]statusCode{
+	trace.StatusCodeOK:                 {message: "ok", httpCode: http.StatusOK},
+	trace.StatusCodeCancelled:          {message: "cancelled", httpCode: 499},
+	trace.StatusCodeUnknown:            {message: "unknown", httpCode: http.StatusInternalServerError},
+	trace.StatusCodeInvalidArgument:    {message: "invalid_argument", httpCode: http.StatusBadRequest},
+	trace.StatusCodeDeadlineExceeded:   {message: "deadline_exceeded", httpCode: http.StatusGatewayTimeout},
+	trace.StatusCodeNotFound:           {message: "not_found", httpCode: http.StatusNotFound},
+	trace.StatusCodeAlreadyExists:      {message: "already_exists", httpCode: http.StatusConflict},
+	trace.StatusCodePermissionDenied:   {message: "permission_denied", httpCode: http.StatusForbidden},
+	trace.StatusCodeResourceExhausted:  {message: "resource_exhausted", httpCode: http.StatusTooManyRequests},
+	trace.StatusCodeFailedPrecondition: {message: "failed_precondition", httpCode: http.StatusBadRequest},
+	trace.StatusCodeAborted:            {message: "aborted", httpCode: http.StatusConflict},
+	trace.StatusCodeOutOfRange:         {message: "out_of_range", httpCode: http.StatusBadRequest},
+	trace.StatusCodeUnimplemented:      {message: "unimplemented", httpCode: http.StatusNotImplemented},
+	trace.StatusCodeInternal:           {message: "internal", httpCode: http.StatusInternalServerError},
+	trace.StatusCodeUnavailable:        {message: "unavailable", httpCode: http.StatusServiceUnavailable},
+	trace.StatusCodeDataLoss:           {message: "data_loss", httpCode: http.StatusNotImplemented},
+	trace.StatusCodeUnauthenticated:    {message: "unauthenticated", httpCode: http.StatusUnauthorized},
 }
 
-func canonicalCodeString(code int32) string {
-	if code < 0 || int(code) >= len(canonicalCodes) {
-		return "error code " + strconv.FormatInt(int64(code), 10)
+type statusCode struct {
```

Comment:
Let's call this `codeDetails` and please add some documentation. e.g.
```go
// codeDetails specifies information about a trace status code.
type codeDetails struct {
    message string // status message
    status  int    // corresponding HTTP status code
}
```
It seems like a more appropriate name.